app/editor/extensions/HoverPreviews.tsx

```diff
@@ -45,8 +49,27 @@ export default class HoverPreviews extends Extension {
               );
               if (isHoverTarget(target, view)) {
                 hoveringTimeout = setTimeout(
-                  action(() => {
-                    this.state.activeLinkElement = target as HTMLElement;
+                  action(async () => {
+                    const element = target as HTMLElement;
+
+                    const url =
+                      element?.getAttribute("href") || element?.dataset.url;
+                    const documentId = window.location.pathname
+                      .split("/")
+                      .pop();
+
+                    if (url) {
+                      try {
+                        const data = await client.post("/urls.unfurl", {
+                          url: url.startsWith("/") ? env.URL + url : url,
+                          documentId,
+                        });
+                        this.state.activeLinkElement = element;
```

Comment:
I haven't ran this yet but setting the `activeLinkElement` state only after the async network request has come back seems like a recipe for buggy behavior…?